lightning-background-processor/src/lib.rs

```diff
@@ -372,7 +372,7 @@ where
 	CF::Target: 'static + chain::Filter,
 	CW::Target: 'static + chain::Watch<<K::Target as SignerProvider>::Signer>,
 	T::Target: 'static + BroadcasterInterface,
-	K::Target: 'static + KeysInterface,
+	K::Target: 'static + EntropySource + NodeSigner + SignerProvider,
```

Comment:
I agree we should split things up, but I would lean against doing all the splitting in this PR, maybe with the exception of some very shallow-level API examples, like BackgroundProcessor. BackgroundProcessor definitely lacks generics.

Reply:
Well, one benefit of splitting it up into two is that it would allow @wpaulino to open his node_secret removal PR against the comment that's currently in this one, while I work on the follow-up in parallel. But I'll work on the parameter split tonight, maybe I can get it in fast enough and it's less trouble than I'm anticipating.